tests/rich_env/serializers/test_base_model.py

```diff
@@ -17,26 +17,44 @@ def initialize(self, module: str):
         self.module = module
         self.registry = DefaultSerializerRegistry()
 
-    def _assert_model_serializer(self, serializer, expected_serializer):
+    def _assert_serializer(self, serializer, expected_serializer):
         self.assertIsInstance(serializer, expected_serializer)
         self.assertTrue(serializer.stable())
         self.assertTrue(serializer.available())
         self.assertIn(self.module, serializer.meta())
         self.assertIn(self.module, serializer.requirements())
 
+    def base_unpack_test(self, model: Any, expected_serializer) -> tempfile.TemporaryDirectory:
+        serializer = self.registry.find_serializer_by_type(type(model))
+
+        assert isinstance(serializer, ModelBaseSerializer)
+        self._assert_serializer(serializer, expected_serializer)
+
+        with tempfile.TemporaryFile() as file:
+            serializer.serialize(model, file)
+            file.flush()
+            file.seek(0)
+
+            self.assertEqual(serializer.data_format(), Serializer.deserialize_data_format(file))
+
+            temp_dir = tempfile.TemporaryDirectory()
+            serializer.unpack_model(file, temp_dir.name)
+            return temp_dir
```

Comment:
I believe that you shoud use `@contextmanager` decorator over `base_unpack_test` and do
```
with tempfile.TemporaryDirectory() as temp_dir:
    yield temp_dir
```
instead of just returning temp_dir to make sure it will be removed in case of inappropriate usage of your method.

Reply:
good point, thx